versions-common/src/main/java/org/codehaus/mojo/versions/utils/MavenProjectUtils.java

```diff
@@ -150,4 +154,18 @@ public static Dependency interpolateVersion(final Dependency dependency, final M
         }
         return dependency;
     }
+
+    /**
+     * Reads the given file to a StringBuilder
+     * @param path file path
+     * @return StringBuilder containing the given file
+     * @throws IOException thrown in case of an I/O error
+     */
+    public static StringBuilder readFile(Path path) throws IOException {
+        try (BufferedReader reader = Files.newBufferedReader(path)) {
+            return reader.lines().collect(StringBuilder::new, StringBuilder::append, StringBuilder::append);
```

Comment:
Does line end character can be important here? 
In old implementation we read bytes from file but here line end character can be drop.

Reply:
Easier to debug if tests go wrong, if I remember correctly.

Reply:
Ok, we are not adding any extra new lines here.

I checked and now I remember that: in the old implementation we are reading into a byte array and then converting that byte array to a String using the default encoding. In some systems, e.g. Windows, this produces wrong results resulting in these "umlauts" being incorrect. 

Using a StringBuilder in between is free from this problem. That is the whole fix to this issue.

Reply:
Still in a draft, but for the record: the issue is caused by a *discrepancy between `-Dfile.encoding` and the actual encoding of the file*. The system property determines the value of the `Charset.defaultCharset()` used to convert the bytes[] read in Files.read() into a string using `new String(byte[])`.

Reply:
So, I don't really know whether this is a bug or not. In 2.14.2 it did not need to use the raw file so it relied completely on Maven to process the file. In version 2.15.0, the plugin processes raw POM and thus is susceptible to problems with encoding.

In this case, the problem is caused by the user having the **wrong value of their -Dfile.encoding** system property, which causes String(byte[]) to use the wrong encoding (given by the user) to convert the byte array to string. This is why I also wasn't able to reproduce the issue.

This same problem will occur everywhere where we process raw pom and a user will have a system with the wrong config.

So, as an attempt to solve this problem -- I'm using Plexus's XmlReader with its detection capability to detect the encoding. Please tell me what you think of this solution.